Add README to Prow Directory

chases2 opened the pull request:
Copied from another Prow repository with a similar bumping strategy

clarketm commented on prow/README.md:
````diff
@@ -0,0 +1,180 @@
+kubectl --context=istio-prow create -f ~/foo.yaml
+```
+
+### Update Config File
````
Remove. This doesn't apply.

clarketm commented on prow/README.md:
````diff
@@ -0,0 +1,180 @@
+$ chmod +x prow/my-presubmit.sh
+```
+
+### Test-Infra Prow Jobs
````
Remove. This doesn't apply.

clarketm commented on prow/README.md:
```diff
@@ -0,0 +1,180 @@
+Then you can access to the artifact through "artifacts" at [gubernator](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/1025/e2e-suite-rbac-no_auth/1006/)
+
+
+### Prow Bazel Build Image
```
Remove. This doesn't apply.

clarketm commented on prow/README.md:
````diff
@@ -0,0 +1,180 @@
+- trigger
+```
+
+### Upload artifacts after job finishes
````
Remove. This doesn't apply.

chases2 commented:
/hold

chases2 commented:
Incorrect documentation can only lead to problems; removed the irrelevant sections from README. The example in `Github Trigger` still references istio, but as an example that still applies to this configuration also.

I think it's worth retaining everything in this new version, but I'm most attached to the manual information in `Upgrading Prow`, which is now unique to this Prow instance.

Also fixed a broken link in the top-level README pointing to config.yaml

chases2 commented:
/hold cancel

chases2 commented:
/assign @clarketm

clarketm commented:
@chases2 - I agree. Thanks for the documentation and cleanup!

clarketm commented:
/lgtm

krzyzacy commented:
@clarketm let me add you to collaborators

clarketm commented:
Great! Thanks @krzyzacy.

krzyzacy commented:
lol am I drunk? I cannot find the access to repo settings anymore?

clarketm commented:
@krzyzacy
![image](https://user-images.githubusercontent.com/9957358/64585258-de2d5980-d34c-11e9-833c-786452ef34aa.png)

krzyzacy commented:
I mean... I obviously don't have that access anymore

krzyzacy commented:
punt to @fejta

clarketm commented:
Oh, I misunderstood. Now, I see what you mean. I can ask @fejta.
